galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.tsq.manager/src/main/java/dev/galasa/cicsts/tsq/internal/TsqImpl.java

```diff
@@ -0,0 +1,285 @@
+			logger.info("Message length written to variable is : " + varLength);
+			logger.info("Message written is : " + inputData);
+
+		} catch (CeciException e) {
+			throw new TsqException("Failed to create variable for input data to write to TSQ : " + this.queueName + ".", e);
+		}
+
+		try {
+			// Write to TSQ using using CECI command
+			ICeciResponse resp = this.ceci.issueCommand(this.ceciTerminal, command);
+			if (!resp.isNormal()) {
+				throw new TsqException("Write to TSQ failed for queue " + this.queueName + ". CICS response: " + resp.getResponse());
+			} else {
+				logger.info(String.format("Written '%s' to TSQ - %s successful.",inputData , this.queueName ));
+			}
+		} catch (CeciException e) {
+			throw new TsqException("Failed to write to TSQ : " + this.queueName + ".", e);
+		} 
+		
+		// Delete the &INPUTDATA variable after writing to TSQ
+		try{
+			this.ceci.deleteVariable(this.ceciTerminal, "&INPUTDATA");
+		} catch (CeciException e) {
+			throw new TsqException("Failed to delete variable used to write to TSQ : " + this.queueName + ".", e);
+		}	
+        return;
+    }          
+
+	/*
+	 * To delete the TSQ with name set using setName() method
+	 */
+    @Override
+    public void deleteQ() throws TsqException { 
+		// Check if queueName is set
+        this.checkQueueName();
+		
+		// Create DELETEQ command to execute using CECI
+		String command = "DELETEQ TS";
+		if(this.queueName.length() > 8) {
+			// Use QNAME where the TSQ name is greater than 8 characters.
+			command += " QNAME(" + this.queueName + ")";
+		} else {
+			// Otherwise use QUEUE to avoid potentially breaking existing tests.
+			command += " QUEUE(" + this.queueName + ")";
+		}
+        
+        try {	
+			//Start CECI session in the terminal
+			this.ceci.startCECISession(this.ceciTerminal);
+            
+			// Delete TSQ using using CECI command
+			ICeciResponse resp = this.ceci.issueCommand(this.ceciTerminal, command);
+			if (!resp.isNormal()) {
+                throw new TsqException("Delete TSQ failed for queue " + this.queueName + ". CICS response: " + resp.getResponse());
+            }
+		} catch(CeciException e) {
+			throw new TsqException("Failed to delete TSQ " + this.queueName + ".", e);
+		}
+		return;
+	}
+
+	/*
+	 * Make recoverable the TSQ with name set using setName() method
+	 */
+	@Override
+	public void makeRecoverable() throws TsqException {
+		// Generate model name 
+        String modelName = resolveModelName();
+		try {
+			// Run CEDA TSMODEL command to make TSQ recoverable
+            this.ceda.createResource(this.cedaTerminal, "TSMODEL", modelName, modelName, String.format("PREFIX(%s) RECOVERY(YES)", this.queueName));
+			// Install CEDA TSMODEL to make TSQ recoverable
+            this.ceda.installResource(this.cedaTerminal, "TSMODEL", modelName, modelName);
+		} catch(CedaException e) {
+			throw new TsqException("Failed to make TSQ : " + this.queueName + " recoverable.", e);
+		}
+		return;
+	}    
```

Comment:
Made recoverable option a parameter for createQueue() method, so that the TSQ will be made recoverable during the first write.
```
String createQueue(String queueName, String data, boolean recoverable); 
// If TSQ needs to be created with recoverable option, Returns OK, ALREADY EXISTING, ERROR.
```

Reply:
Design changed to have a TsqFactory to createQueue() with queue name as parameter and return a ITsq object for the queue name.